engine/src/test/java/io/camunda/zeebe/engine/state/distribution/DistributionStateTest.java

```diff
@@ -0,0 +1,167 @@
+import io.camunda.zeebe.protocol.record.ValueType;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+@ExtendWith(ProcessingStateExtension.class)
+public final class DistributionStateTest {
+  private MutableProcessingState processingState;
+  private MutableDistributionState distributionState;
+
+  @BeforeEach
+  public void setUp() {
+    distributionState = processingState.getDistributionState();
+  }
+
+  @Test
+  public void shouldReturnFalseOnEmptyStateForHasPendingCheck() {
+    // when
+    final var hasPending = distributionState.hasPendingDistribution(10L);
+
+    // then
+    assertThat(hasPending).isFalse();
+  }
+
+  @Test
+  public void shouldAddPendingDistribution() {
+    // given
+    final var distributionKey = 10L;
+    final var partition = 1;
+
+    // when
+    distributionState.addPendingDistribution(distributionKey, partition);
+
+    // then
+    assertThat(distributionState.hasPendingDistribution(distributionKey)).isTrue();
+  }
+
+  @Test
+  public void shouldRemovePendingDistribution() {
+    // given
+    final var deploymentKey = 10L;
+    final var partition = 1;
+    distributionState.addPendingDistribution(deploymentKey, partition);
+
+    // when
+    distributionState.removePendingDistribution(deploymentKey, partition);
+
+    // then
+    assertThat(distributionState.hasPendingDistribution(deploymentKey)).isFalse();
+  }
+
+  @Test
+  public void shouldReturnNullOnRequestingStoredDistributionWhenNothingStored() {
+    // when
+    final var distributionRecord = distributionState.getCommandDistributionRecord(1, 1);
+
+    // then
+    assertThat(distributionRecord).isNull();
+  }
+
+  @Test
+  public void shouldStoreDistributionInState() {
+    // given
+    final var distributionRecord = createCommandDistributionRecord();
+
+    // when
+    distributionState.addCommandDistribution(1, distributionRecord);
+
+    // then
+    final var storedDistribution =
+        distributionState.getCommandDistributionRecord(1, distributionRecord.getPartitionId());
+
+    assertThat(storedDistribution).isNotNull().isEqualTo(distributionRecord);
+  }
+
+  @Test
+  public void shouldRemoveDistribution() {
+    // given
+    final var distributionRecord = createCommandDistributionRecord();
+    distributionState.addCommandDistribution(1, distributionRecord);
+
+    // when
+    distributionState.removeCommandDistribution(1);
+
+    // then
+    final var storedDistribution =
+        distributionState.getCommandDistributionRecord(1, distributionRecord.getPartitionId());
+    assertThat(storedDistribution).isNull();
+  }
+
+  @Test
+  public void shouldRemoveDifferentDistributions() {
+    // given
+    final var distributionRecord = createCommandDistributionRecord();
+    distributionState.addCommandDistribution(1, distributionRecord);
+    distributionState.addCommandDistribution(2, distributionRecord);
```

Comment:
🔧 This test confuses me a bit. Perhaps it would be better fitting if the two command distributions (with different keys) are actually different distribution records.